src/citrine/builders/predictors.py

```diff
@@ -0,0 +1,128 @@
+from typing import Tuple, List, Optional
+
+from citrine.resources.project import Project
+from citrine.informatics.predictors import (
+    MolecularStructureFeaturizer,
+    MeanPropertyPredictor
+)
+from citrine.informatics.descriptors import Descriptor, FormulationDescriptor, RealDescriptor
+
+
+def mean_feature_properties(
+        *,
+        project: Project,
+        featurizer: MolecularStructureFeaturizer,
+        formulation_descriptor: FormulationDescriptor,
+        p: int,
+        impute_properties: bool = True,
+        all_ingredients: bool = True,
+        labels: Optional[List[str]] = None
+) -> Tuple[List[MeanPropertyPredictor], List[Descriptor]]:
+    """[ALPHA] Combine a featurizer model with mean property models.
+
+    Given a featurizer, produce "mean property" models that calculate the mean of the
+    properties calculated by the featurizer. This is useful if you do not directly know
+    the numerical properties of ingredients in a formulation, but instead know, for example,
+    the molecular structure. This builder method determines the real-valued properties that the
+    featurizer calculates and builds mean property models that use them as input properties.
+
+    Parameters
+    ----------
+    project: Project
+        Project that contains the predictor
+    featurizer: MolecularStructureFeaturizer
+        A model that is being used to featurize formulation ingredients. Currently only
+        accepts a molecular structure featurizer, but other models can be used in principle.
+        TODO: add chemical formula featurizer once it is exposed
+    formulation_descriptor: FormulationDescriptor
+        Descriptor that represents the formulation being featurized.
+    p: int
+        Power of the generalized mean. Only integer powers are supported.
+    impute_properties: bool
+        Whether to impute missing ingredient properties by averaging over the entire dataset.
+        If ``False``, an error is thrown if any ingredient being featurized is missing a property.
+    all_ingredients: bool
```

Comment:
Maybe rename this to `featurize_all_ingredients` or something similar? I'm not sure what it's doing to `all_ingredients`. Also maybe describe what happens if it's false: `If False, MPP's are only constructed for the specified labels. Must be True if no Labels are specified.`